app/src/main/java/kr/ac/konkuk/gdsc/hyeseon/presentation/mypage/EditViewModel.kt

```diff
@@ -0,0 +1,28 @@
+package kr.ac.konkuk.gdsc.hyeseon.presentation.mypage
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import kr.ac.konkuk.gdsc.hyeseon.util.code.ErrorCode
+import kr.ac.konkuk.gdsc.hyeseon.util.view.InputUiState
+
+class EditViewModel : ViewModel() {
+
+    val _editableNickName = MutableStateFlow("")
+    val editableNickName: String get() = _editableNickName.value
+    val isValidNickName = MutableSharedFlow<InputUiState>()
```

Comment:
```suggestion
    val isValidNickName = _editableNickName.map {
        // 비어있는 경우 Failure state
        // 조건에 맞으면 Success
    }
```

이런식으로 입력값을 검증해서 새로운 Flow로 반환해주는 Reactive 프로그래밍 방식을 사용하는것이 Flow를 사용하면서 활용해야하는 사고방식(능력)을 증진시키는데 도움이 됩니다.